python/lsst/afw/display/rgb.py

```diff
@@ -32,34 +88,55 @@ def makeRgbImage(self, imageR, imageG, imageB):
             if hasattr(c, "getArray"):
                 imageRGB[i] = c.getArray()
 
-        return np.flipud(np.dstack(self._convertImagesToUint8(*imageRGB)).astype(np.uint8))
+        if xSize is not None or ySize is not None:
+            assert rescaleFactor is None, "You may not specify a size and rescaleFactor"
+            h, w = imageRGB[0].shape
+            if ySize is None:
+                ySize = int(xSize*h/float(w) + 0.5)
+            elif xSize is None:
+                xSize = int(ySize*w/float(h) + 0.5)
+
+            size = (ySize, xSize) # n.b. y, x order for scipy
+        elif rescaleFactor is not None:
+            size = float(rescaleFactor) # an int is intepreted as a percentage
+        else:
+            size = None
+
+        if size is not None:
+            try:
+                import scipy.misc
+            except ImportError as e:
+                raise RuntimeError("Unable to rescale as scipy.misc is unavailable: %s" % e)
+
+            for i, im in enumerate(imageRGB):
+                imageRGB[i] = scipy.misc.imresize(im, size, interp='bilinear', mode='F')
+
+        return np.dstack(self._convertImagesToUint8(*imageRGB)).astype(np.uint8)
 
     def intensity(self, imageR, imageG, imageB):
-        """!Return the total intensity from the red, blue, and green intensities"""
-        return (imageR + imageG + imageB)/float(3);
+        """!Return the total intensity from the red, blue, and green intensities
 
-    def mapIntensityToUint8(self, I):
-        """Map an intensity into the range of a uint8, [0, 255] (but not converted to uint8)"""
-        return np.where(I <= 0, 0, np.where(I < self._uint8Max, I, self._uint8Max))
+        This is a naive computation, and may be overridden by subclasses
+        """
+        return computeIntensity(imageR, imageG, imageB)
 
     def _convertImagesToUint8(self, imageR, imageG, imageB):
         """Use the mapping to convert images imageR, imageG, and imageB to a triplet of uint8 images"""
```

Comment:
It would be helpful to add a comment here that subclasses must define mapIntensityToUint8, since there is no longer a default.